uktrade/data-hub-frontend
# Integrate Great.gov.uk form(s) into Data Hub

## Description of change

This piece of work integrates Great.gov.uk forms into Data Hub, initially it's just the `Export enquiry` form, more will be added over time. The new functionality is hidden behind a feature flag called `activity-feed-export-enquiry` which is currently turned on in `dev` and `staging` but off in `prod.`

To return `Export enquiry` documents from Elasticsearch four `terms` must match:

**1. Ensures we only match Great.gov.uk forms (which there are many)**
```js
{
   term: {
     'object.type': 'dit:directoryFormsApi:Submission'
   }
}
```

**2. Ensures the document returned is in JSON format (there are two other formats HTML and Text)**
```js
{
  term: {
    'object.attributedTo.type': 'dit:directoryFormsApi:SubmissionAction:gov-notify-email',
   }
}
````

**3. Ensures we only match Export enquires**
This is the great.gov.uk path of the form _(e.g. great.gov.uk/contact/export-advice/comment)_
```js
{
  term: {
    'object.url': '/contact/export-advice/comment/'
  }
}
```

**4. Ensures we match a Data Hub company contact to the user filling out the form at Great.gov.uk**
```js
{
  terms: {
    'actor.dit:emailAddress': [
      ...contacts.map((contact) => contact.email),
    ]
  }
}

```

It's far easier to read and maintain our own Elasticsearch queries compared to using `bodybuilder` (an ES query builder) which I've removed due to its inflexibility.

## Test instructions

Bring up the FE locally and point to the dev API on PaaS, then go to: `/companies/6df487c5-7c75-4672-8907-f74b49e6c635/activity` and apply the `All external activity` filter. 

You should see two external activities:

<img width="969" alt="great gov uk-activity" src="https://user-images.githubusercontent.com/964268/101015799-d7958e00-355f-11eb-8464-2c3258e3e38b.png">

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
